core/routemgmt/activateApi.js

```diff
@@ -0,0 +1,383 @@
+  })
+  .catch(function(reason) {
+      if (!gwApiActivated) {
+        console.error('API activation failure: '+reason);
+        return Promise.reject('API activation failure: '+reason);
+      }
+      console.error('API was activated; however, the database could not be udpated: '+reason);
+      return Promise.reject('API was activated; however, an internal error occurred after activation: '+reason);
+  });
+}
+
+function getCloudantAccount(message) {
+  // full cloudant URL - Cloudant NPM package has issues creating valid URLs
+  // when the username contains dashes (common in Bluemix scenarios)
+  var cloudantUrl;
+
+  if (message.url) {
+    // use bluemix binding
+    cloudantUrl = message.url;
+  } else {
+    if (!message.host) {
+      whisk.error('cloudant account host is required.');
+      return;
+    }
+    if (!message.username) {
+      whisk.error('cloudant account username is required.');
+      return;
+    }
+    if (!message.password) {
+      whisk.error('cloudant account password is required.');
+      return;
+    }
+    if (!message.port) {
+      whisk.error('cloudant account port is required.');
+      return;
+    }
+    if (!message.protocol) {
+      whisk.error('cloudant account protocol is required.');
+      return;
+    }
+
+    cloudantUrl = message.protocol + "://" + message.username + ":" + message.password + "@" + message.host + ":" + message.port;
+  }
+
+  return require('cloudant')({
+    url: cloudantUrl
+  });
+}
+
+// Retrieve the specific API document from the DB
+// This is a wrapper around an action invocation (getApi)
+function getApiDoc(namespace, basepath, docid) {
+  var actionName = '/whisk.system/routemgmt/getApi';
+  var params = {
+    'namespace': namespace,
+    'basepath': basepath
+  }
+  if (docid) params.docid = docid;
+  console.log('getApiDoc() for: ', params);
+    return whisk.invoke({
+      name: actionName,
+      blocking: true,
+      parameters: params
+    })
+    .then(function (activation) {
+      console.log('whisk.invoke('+actionName+', '+params.namespace+', '+params.basepath+') ok');
+      console.log('Results: '+JSON.stringify(activation));
+    if (activation && activation.result && activation.result.apis &&
+        activation.result.apis.length > 0 && activation.result.apis[0].value &&
+        activation.result.apis[0].value._rev) {
+        return Promise.resolve(activation.result.apis[0].value);
+      } else {
+        console.error('Invalid API doc returned!');
+        return Promise.reject('Document for namepace \"'+namespace+'\" and basepath \"'+basepath+'\" was not located');
+      }
+    })
+    .catch(function (error) {
+      console.error('whisk.invoke('+actionName+', '+params.namespace+', '+params.basepath+') error: '+JSON.stringify(error));
+      return Promise.reject(error);
+    });
+}
+
+function addTenantToGateway(gwInfo, namespace) {
+  var options = {
+    url: gwInfo.gwUrl+'/tenants',
+    agentOptions: {rejectUnauthorized: false},
```

Comment:
Good observation.  I just checked with the API GW folks and they now use a CA signed cert, so this limitation can be removed.